src/main/scala/viper/gobra/translator/implementations/components/ArraysImpl.scala

```diff
@@ -0,0 +1,137 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2020 ETH Zurich.
+
+package viper.gobra.translator.implementations.components
+
+import viper.gobra.translator.interfaces.Collector
+import viper.gobra.translator.interfaces.components.Arrays
+import viper.silver.ast.DomainFuncApp
+import viper.silver.{ast => vpr}
+
+/**
+  * Contains functionality for generating and using the "Array" Viper domain.
+  */
+class ArraysImpl extends Arrays {
```

Comment:
Is this class still needed? Isn't it superseded by [Exclusive|Shared]ArrayComponent?

Reply:
We might need it for another encoding, so I would just leave it.